File: math/code/inverse/inverse-2.py
# --8<-- [start:core]
# Returns the modular inverse of a prime modulo p.
# Use built-in pow function.
def inverse(a, p):
    return pow(a, p - 2, p)


# --8<-- [end:core]

if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        a, p = map(int, input().split())
        print(inverse(a, p))

File: math/code/inverse/inverse-2.cpp
#include <iostream>

// --8<-- [start:core]
// Binary exponentiation.
int pow(int a, int b, int m) {
  long long res = 1, po = a;
  for (; b; b >>= 1) {
    if (b & 1) res = res * po % m;
    po = po * po % m;
  }
  return res;
}

// Returns the modular inverse of a prime modulo p.
int inverse(int a, int p) { return pow(a, p - 2, p); }

// --8<-- [end:core]
int main() {
  int t;
  std::cin >> t;
  for (; t; --t) {
    int a, p;
    std::cin >> a >> p;
    std::cout << inverse(a, p) << std::endl;
  }
}
